packages/jest/src/utils/config/is-preset-cjs.ts

```diff
@@ -0,0 +1,12 @@
+import { type Tree, readJson } from '@nx/devkit';
+
+export function isPresetCjs(tree: Tree) {
+  if (tree.exists('jest.preset.cjs')) {
+    return true;
+  }
+
+  const rootPkgJson = readJson(tree, 'package.json');
+  if (rootPkgJson.type && rootPkgJson.type === 'module') {
+    return true;
+  }
```

Comment:
Not sure it's by design but not all paths return a boolean in this function, there's no default `return false` branch, which could lead to bug if used this way `isPresetCjs(tree) === false`.

Reply:
It’s a good catch, it should be updated.